engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/CallCache.scala

```diff
@@ -0,0 +1,40 @@
+package cromwell.engine.workflow.lifecycle.execution.callcaching
+
+import cromwell.backend.BackendJobExecutionActor.SucceededResponse
+import cromwell.core.{JobOutputs, WorkflowId}
+import cromwell.database.sql._
+import cromwell.database.sql.tables.CallCachingResultMetainfoEntry
+import cromwell.engine.workflow.lifecycle.execution.callcaching.EngineJobHashingActor.CallCacheHashes
+import cromwell.core.ExecutionIndex.IndexEnhancedIndex
+
+import scala.concurrent.{ExecutionContext, Future}
+
+/**
+  * Given a database-layer CallCacheStore, this accesser can access the database with engine-friendly data types.
+  */
+class CallCache(database: CallCachingStore) {
+  def addToCache(workflowId: WorkflowId, callCacheHashes: CallCacheHashes, response: SucceededResponse)(implicit ec: ExecutionContext): Future[Unit] = {
+    val metainfo = CallCachingResultMetainfoEntry(
+      workflowUuid = workflowId.toString,
+      callFqn = response.jobKey.call.fullyQualifiedName,
+      scatterIndex = response.jobKey.index.fromIndex,
+      returnCode = response.returnCode,
+      allowResultReuse = true,
```

Comment:
Actually as a **parameter**, I think probably not. At least not with our current set of options...